### ISSUE TITLE
add `prefer-enum-over-literal-union` rule

### ISSUE THREAD

**Issue by marcalexiei** [marcalexiei/eslint-plugin-zod-x#181]
### 🧩 Description

To ensure feature parity with `eslint-plugin-zod` we should implement a `prefer-enum-over-literal-union`

Code: https://github.com/gajus/eslint-plugin-zod/blob/main/src/rules/preferEnum.ts
Test: https://github.com/gajus/eslint-plugin-zod/blob/main/test/rules/assertions/preferEnum.ts

### ⚙️ Options

No options

### 🧾 Example Configuration

No options

### ✅ Valid Code Examples

```typescript
z.enum(['foo', 'bar'])
```

### ❌ Invalid Code Examples

```typescript
z.union([z.literal('foo'), z.literal('bar')])
```
